extension/jaegerremotesampling/internal/http.go

```diff
@@ -0,0 +1,122 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/jaegerremotesampling/internal"
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/jaegertracing/jaeger/cmd/agent/app/configmanager"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+)
+
+var (
+	errMissingClientConfigManager = errors.New("the client config manager has not been provided")
+)
+
+var _ component.Component = (*SamplingHTTPServer)(nil)
+
+type SamplingHTTPServer struct {
+	telemetry component.TelemetrySettings
+	settings  confighttp.HTTPServerSettings
+	cfgMgr    configmanager.ClientConfigManager
+
+	mux        *http.ServeMux
+	srv        *http.Server
+	shutdownWG *sync.WaitGroup
+}
+
+func NewHTTP(telemetry component.TelemetrySettings, settings confighttp.HTTPServerSettings, cfgMgr configmanager.ClientConfigManager) (*SamplingHTTPServer, error) {
```

Comment:
At this point, we could return, as the caller of this does expect a component.Component. The tests are currently benefiting from this being a concrete type. How about we do a refactoring session once we see the whole solution ready? I think this will make sense in the next steps, but I don't want to optimize it just now.